openmc/stats/multivariate.py

```diff
@@ -613,6 +616,117 @@ def from_xml_element(cls, elem):
         origin = [float(x) for x in elem.get('origin').split()]
         return cls(r, phi, z, origin=origin)
 
+class MeshSpatial(Spatial):
+    """Spatial distribution for a mesh.
+
+    This distribution specifies a mesh to sample over, chooses whether it will be volume normalized, and can set the source strengths. 
+
+    .. versionadded:: 0.13
+
+    Parameters
+    ----------
+    mesh : openmc.MeshBase
+        The mesh instance used for sampling, mesh is written into settings.xml, mesh.id is written into the source distribution
+    strengths : Iterable of Real, optional
+        A list of values which represent the weights of each element
+
+
+    Attributes
+    ----------
+    mesh : openmc.MeshBase
+        The mesh instance used for sampling, mesh is written into settings.xml, mesh.id is written into the source distribution
+    strengths : Iterable of Real, optional
+        A list of values which represent the weights of each element
+
+    """
+
+    def __init__(self, mesh, strengths=None, volume_normalized=True):
+        self.mesh = mesh        
+        self.strengths = strengths
+        self.volume_normalized = volume_normalized
+
+    @property
+    def mesh(self):
+        return self._mesh
+
+    @mesh.setter
+    def mesh(self, mesh):
+        if mesh != None:
+            cv.check_type('Unstructured Mesh instance', mesh, MeshBase)
+        self._mesh = mesh
+
+    @property
```

Comment:
part of me wonders if it is more straightforward to drop the `volume_normalized` stuff in favor of something like `mesh_spatial_obj.strengths = strengths*mesh.volumes` or something. This forces the user to know what they are doing when setting the `strengths` property.

Reply:
That would be nice, but for unstructured meshes we don't know the element volumes until runtime. :-/

We could extract this info with the CAPI, but I think that would unecessarily complicate any analysis workflows using this capability.

Reply:
mmm pesky. Is it worth adding to `UnstructuredMesh` in the Python API? It may help for metadata manipulation later on to have more functionality on the python side generally speaking.

Reply:
We _can_ query element volumes from the `UnstructuredMesh` class, but it has to be loaded from a statepoint file. This is similar to tallies in that they don't have any data unless they're loaded from a statepoint. 

Being able to query element connectivity, vertex coordinates, etc. before running a simulation would be challenging and require us to support some interface to NetCDF files -- a nontrivial lift I'm afraid 😞